lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -118,6 +123,33 @@ static Optional<APSInt> getExtendedConstant(Value operand, Attribute constant,
   return {};
 }
 
+/// Determine the value of a constant operand for the sake of constant folding.
+/// This will map `invalidvalue` to a zero value of the corresopnding type,
+/// which aligns with how the Scala FIRRTL compiler handles invalids.
+static Optional<APSInt> getConstant(Attribute operand) {
+  if (!operand)
+    return {};
+  if (auto attr = operand.dyn_cast<InvalidValueAttr>()) {
+    if (auto type = attr.getType().dyn_cast<IntType>())
+      return APSInt(type.getWidth().getValueOr(1), type.isUnsigned());
```

Comment:
Can `getWidth().getValue()` fail here? I assume this is catching the case of this running on an uninferred width?

Reply:
Yeah exactly. My thinking was that there's this `getAPSInt()` call below if we're dealing with an `IntegerAttr`, which would have a width assigned even in the case that the FIRRTLType itself had an uninferred width (just some random width that is sufficient to hold the constant). So in analogy I just made the invalidvalue fold to a zero of at least one bit.